ui/app/components/totp-edit.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import RoleEdit from './role-edit';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+
+export default class TotpEdit extends RoleEdit {
+  @tracked hasGenerated = false;
+  successCallback;
+
+  init() {
+    super.init(...arguments);
+    this.set('backendType', 'totp');
+  }
+
+  persist(method, successCallback) {
+    const model = this.model;
+    return model[method]().then(() => {
+      if (!model.isError) {
+        if (model.backend === 'totp' && model.generate) {
+          this.hasGenerated = true;
+          this.successCallback = successCallback;
+        } else {
+          successCallback(model);
+        }
+      }
+    });
+  }
```

Comment:
Writing a delete method specific for this item should remove the need for some conditionals here. But I'd still like to confirm what the intended functionality is here. Is this called when clicking "Delete" from the popup menu?